File: src/main/java/com/footverse/auth/dto/ForgotPasswordRequest.java
package com.footverse.auth.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;

/**
 * Requests a password-reset one-time code by email (dto-spec §6). The response is always the same
 * whether or not the email belongs to an enabled account (business-rules → Authentication).
 *
 * @param email the account's email
 */
public record ForgotPasswordRequest(
        @NotBlank @Email String email) {
}

File: app/lib/features/auth/models/forgot_password_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'forgot_password_request.g.dart';

/// Request a password-reset one-time code by email (dto-spec §6).
@JsonSerializable(createFactory: false)
class ForgotPasswordRequest {
  const ForgotPasswordRequest({required this.email});

  final String email;

  Map<String, dynamic> toJson() => _$ForgotPasswordRequestToJson(this);
}
